app/forms/register_user_email_form.rb

```diff
@@ -40,8 +41,15 @@ def submit(params, instructions = nil)
   attr_writer :email
   attr_reader :success
 
-  def valid_form?
-    @allow && valid? && !email_taken?
+  def valid_form?(request_id)
```

Comment:
Since the forms include `ActiveModel::Model`, any reason not to add an active model validation here? E.g.

```
validate :service_provider_request_exists

# ...

def service_provider_request_exists
  return if ServiceProviderRequest.where(uuid: request_id).any?
  errors.add(:email, t('sign_up.email.invalid_request'))
end
```